internals/api/encrypted_value.go

```diff
@@ -0,0 +1,93 @@
+package api
+
+import (
+	"encoding/json"
+
+	"github.com/secrethub/secrethub-go/internals/crypto"
+)
+
+var (
+	ErrInvalidEncryptionType = errAPI.Code("invalid_encryption_type").Error("invalid encryption method provided for key")
+)
+
+type EncryptionType string
+
+const (
+	EncryptionTypeRSAAES  EncryptionType = "rsa-oaep-aes-gcm"
+	EncryptionTypeAWSKKMS EncryptionType = "aws-kms"
+)
+
+type EncryptedValue struct {
```

Comment:
Shall we add some godoc explaining these types and their fields?

Reply:
Jepps. If the overall structure is okay, I'll take some time to fix that.